### PR TITLE
[Feat] add histogram support

### PR DESCRIPTION
**Pull request by pepoviola**
- Add histogram syntax

```
<node name>: reports histogram<metric name, with labels for match> has at least <x> samples in buckets <[array of buckets]> [within x seconds]
```

e.g: 
```
alice: reports histogram polkadot_pvf_execution_time has at least 2 samples in buckets ["0.1", "0.05", "0.025", "+Inf"] within 100 seconds
```

cc @sandreim

### CODE REVIEW

File: src/networkNode.ts [hunk 1]
```diff
@@ -254,6 +254,79 @@ export class NetworkNode implements NetworkNodeInterface {
...
+        clearTimeout(limitTimeout);
+        return value;
+      }
+
+      // loop until get the desired value or timeout
+      let done = false;
+      let c = 0;
+      while (!done) {
+        if (expired) throw new Error(`Timeout(${timeout}s)`);
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (sandreim) [inline src/networkNode.ts:286]: We should have a constant for this.

File: src/networkNode.ts [hunk 2]
```diff
@@ -254,6 +254,79 @@ export class NetworkNode implements NetworkNodeInterface {
...
+      if (desiredMetricValue === null || value >= desiredMetricValue) {
+        debug(`value: ${value} ~ desiredMetricValue: ${desiredMetricValue}`);
+        clearTimeout(limitTimeout);
+        return value;
+      }
+
+      // loop until get the desired value or timeout
+      let done = false;
+      let c = 0;
+      while (!done) {
```
Comment (sandreim) [inline src/networkNode.ts:284]: Would this scale well when we have grouping and 100+ nodes ?

File: tests/0010-histogram-smoke-test.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[settings]
+timeout = 1000
+bootnode = true
+# provider = "Podman"
+
+[relaychain]
+default_image = "{{ZOMBIENET_INTEGRATION_TEST_IMAGE}}"
+chain = "rococo-local"
+#default_command = "../polkadot/target/release/polkadot"
```
Comment (sandreim) [inline tests/0010-histogram-smoke-test.toml:9]: This is currently a pain to switch between CI and local tests

**Review by sandreim — APPROVED** [review]

File: src/networkNode.ts (hunk 1 above)
Comment (pepoviola) [reply src/networkNode.ts:286]: Yes, sounds good.

File: src/networkNode.ts (hunk 2 above)
Comment (pepoviola) [reply src/networkNode.ts:284]: We will need to change some of the logic for `group` testing, is part of the refactoring that I'm working.
Thx!

**Comment by pepoviola** [comment]
closes https://github.com/paritytech/zombienet/issues/89

**Pull request merged**